src/cargo/ops/cargo_add/dependency.rs

```diff
@@ -0,0 +1,948 @@
+            self.source.as_ref(),
+            self.registry.as_ref(),
+            self.rename.as_ref(),
+        ) {
+            // Extra short when version flag only
+            (
+                false,
+                None,
+                true,
+                Some(Source::Registry(RegistrySource { version: v })),
+                None,
+                None,
+            ) => toml_edit::value(v),
+            // Other cases are represented as an inline table
+            (_, _, _, _, _, _) => {
+                let mut table = toml_edit::InlineTable::default();
+
+                match &self.source {
+                    Some(Source::Registry(src)) => {
+                        table.insert("version", src.version.as_str().into());
+                    }
+                    Some(Source::Path(src)) => {
+                        let relpath = path_field(crate_root, &src.path);
+                        if let Some(r) = src.version.as_deref() {
+                            table.insert("version", r.into());
+                        }
+                        table.insert("path", relpath.into());
+                    }
+                    Some(Source::Git(src)) => {
+                        table.insert("git", src.git.as_str().into());
+                        if let Some(branch) = src.branch.as_deref() {
+                            table.insert("branch", branch.into());
+                        }
+                        if let Some(tag) = src.tag.as_deref() {
+                            table.insert("tag", tag.into());
+                        }
+                        if let Some(rev) = src.rev.as_deref() {
+                            table.insert("rev", rev.into());
+                        }
+                    }
+                    None => {}
+                }
+                if table.contains_key("version") {
+                    if let Some(r) = self.registry.as_deref() {
+                        table.insert("registry", r.into());
+                    }
+                }
+
+                if self.rename.is_some() {
+                    table.insert("package", self.name.as_str().into());
+                }
+                if let Some(v) = self.default_features {
+                    table.insert("default-features", v.into());
+                }
+                if let Some(features) = self.features.as_ref() {
+                    let features: toml_edit::Value = features.iter().cloned().collect();
+                    table.insert("features", features);
+                }
+                if let Some(v) = self.optional {
+                    table.insert("optional", v.into());
+                }
+
+                toml_edit::value(toml_edit::Value::InlineTable(table))
+            }
+        };
+
+        table
+    }
+
+    /// Modify existing entry to match this dependency
+    pub fn update_toml(&self, crate_root: &Path, item: &mut toml_edit::Item) {
+        #[allow(clippy::if_same_then_else)]
```

Comment:
clippy are not cargo anymore 😆 #10438
```suggestion
        #[allow(clippy::if_same_then_else)]
```